prime-dotnet-webapi/Services/SiteService.cs

```diff
@@ -37,17 +37,33 @@ public async Task<bool> PecAssignableAsync(int siteId, string pec)
                 .Where(site => site.Id == siteId)
                 .Select(site => new
                 {
-                    site.CareSettingCode,
-                    site.PEC
+                    site.PEC,
+                    healthAuthorityId = (int?)(site as HealthAuthoritySite).HealthAuthorityOrganizationId
                 })
                 .SingleAsync();
 
-            if (siteDto.CareSettingCode == (int)CareSettingType.HealthAuthority
-                || siteDto.PEC == pec)
+            if (siteDto.PEC == pec)
             {
                 return true;
             }
 
+            if (siteDto.healthAuthorityId.HasValue)
+            {
+                var communitySites = await _context.CommunitySites
+                    .Where(cs => cs.PEC == pec)
+                    .AnyAsync();
+
+                var otherHealthAuthoritySites = await _context.HealthAuthoritySites
+                    .AsNoTracking()
+                    .Where(
+                        s => s.PEC == pec
+                        && s.HealthAuthorityOrganizationId != siteDto.healthAuthorityId
+                        )
+                    .AnyAsync();
+
+                return !communitySites && !otherHealthAuthoritySites;
```

Comment:
So this totally works, but it evokes one of my favorite phrases: "there is no trap so deadly as the one you set for yourself".

Do you see the trap?

_What happens if we add a third type of site?_

This breaks because we won't be checking the new type of site. And we don't want to have to remember to come back and update this method of we do add another type of site.
I'm cool with the PR as is, I will probably come back and refactor this

Reply:
I see, I can update that to retrieve all non HA sites